src/server/api/routers/eventRoute.ts

```diff
@@ -3,69 +3,75 @@ import { z } from "zod";
 import { createTRPCRouter, publicProcedure, protectedProcedure } from "../trpc";
 
 import { PrismaClient } from "@prisma/client";
-import Email from "next-auth/providers/email";
+import { TRPCError, initTRPC } from '@trpc/server';
 
 const prisma = new PrismaClient();
 
+type User = {
+    id: string
+    name: string
+    email: string
+    emailVerified: Date
+    image: string
+    createdAt: Date
+    updatedAt: Date
+  }
+
+type Event = {
+    id: string
+    occuringAt: Date
+    ownerID: string
+    address: string
+    createdAt: Date
+    updatedAt: Date
+}
+
+type Participate = {
+    eventID: string
+    userID: Date
+    timeSlots: JSON
+}
+
 export const eventRouter = createTRPCRouter({
     /*
         async function that creates a new event.
-        First check existance of user according to the email.
-        Then, create new event, participate and update event list.
-        Finally return the updated host of event.
+        First check existence of user according to the email.
+        Then, create new event, participate.
     */
     createEvent: protectedProcedure
         .input(
             z.object({
-                occuringAt: z.string().datetime(), //"2020-01-01T00:00:00Z"
-                owner: z.string(),
-                ownerID: z.string().uuid(),
+                email: z.string().email(),
                 address: z.string(),
-                participant: z.array(z.string()),
-                email: z.string().email()
             })
         )
         .mutation(async (req) => {
-            const userCheck = await prisma.User.findUnique({
+            const userCheck : User | null = await prisma.user.findUnique({
                 where: {
                     email: req.input.email,
                 },
             });
             if (!userCheck) {
-                throw new Error("User does not exist");
+                throw new TRPCError({
+                    code: 'FORBIDDEN',
+                    message: 'User does not exist',
+                  });
```

Comment:
This should at least be a `404` not found, instead of a `403` forbidden. 403's are for when the request is valid but the user does not have valid credentials (permissions) for the request